### PR TITLE
Ensure column-major order for matrices and fix HMM_Translate

### LINKED ISSUE #22: Smooth Transition for GLM to Handmade Math
@kiljacken @bvisness Someone is having a issue with handmade-math there's no reason they should have to transpose there matrices when submitting them to OpenGL right? It seem like the transition from GLM to Handmade Math is not as easy as i thought it was.

Here's the Twitter thread to that kinda explains whats going on, on his side, this shows the complete conversation: https://twitter.com/JesseRMeyer/status/768281652682366976

Anyone have any idea whats going on?

### PR DESCRIPTION
**Pull request by bvisness**
Most of the changes in this patch are cosmetic (such as looping by
columns first.) HMM_Translate was incorrectly producing row-major
matrices, which should now be fixed.

Fixes #22

### CODE REVIEW

**Comment by bvisness** [comment]
@JesseRMeyer @StrangeZak Hopefully this will fix the issues from #22. My project still works fine but you should both try with your own projects and let me know how it goes.

File: HandmadeMath.h [hunk 1]
```diff
@@ -987,16 +990,6 @@ HMM_Mat4d(float Diagonal)
...
-    int Rows;
-    for(Rows = 0; Rows < 4; ++Rows)
-    {
-        int Columns;
-        for(Columns = 0; Columns < 4; ++Columns)
-        {
-            Result.Elements[Rows][Columns] = 0.0f;
-        }
-    }
-
```
Comment (bvisness) [inline HandmadeMath.h]: `HMM_Mat4()` already zeros out the matrix, so this was unnecessary.

**Comment by StrangeZak** [comment]
Ill throw it in my project, will report back in a few minutes with a status on if anything changed.

**Comment by StrangeZak** [comment]
Ran fine, not sure if this fixed the issue when converting from GLM, but this is still a good bug fix.

**Pull request merged**